fury/ui/elements.py

```diff
@@ -3052,3 +3054,396 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+            icon_files.append((mode+"-pressed", read_viz_icons(style="new_icons", fname=fname[1])))
+            btn = Button2D(icon_fnames=icon_files)
+
+            def mode_selector(i_ren, _obj, btn):
+                self.current_mode = btn.icon_names[0]
+                i_ren.force_render()
+
+            btn.on_left_mouse_button_pressed = mode_selector
+
+            self.mode_panel.add_element(btn, btn_pos+padding)
+            btn_pos[0] += btn.size[0]+padding
+
+        self.canvas.add_element(self.mode_panel, (0, 0))
+
+        self.mode_text = TextBlock2D(text="Select appropriate drawing mode using below icon")
+        self.canvas.add_element(self.mode_text, (0.0, 0.95))
+
+    def _get_actors(self):
+        """Get the actors composing this UI component."""
+        return self.canvas.actors
+
+    def _add_to_scene(self, scene):
+        """Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        scene : scene
+
+        """
+        self.current_scene = scene
+        self.canvas.add_to_scene(scene)
+
+    def _get_size(self):
+        return self.canvas.size
+
+    def _set_position(self, coords):
+        """Set the lower-left corner position of this UI component.
+
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        self.canvas.position = coords
+
+    def resize(self, size):
+        """Resize the UI.
+        """
+        pass
+
+    @property
+    def current_mode(self):
+        return self._current_mode
+
+    @current_mode.setter
+    def current_mode(self, mode):
+        self.update_button_icons(mode)
+        self._current_mode = mode
+        if mode is not None:
+            self.mode_text.message = f"Mode: {mode}"
+
+    def cal_min_boundary_distance(self, position):
+        """Calculate the minimum distance between the current position and canvas boundary.
+
+        Parameters
+        ----------
+        position: (float,float)
+            current position of the shape.
+
+        Returns
+        -------
+        float
+            Minimum distance from the boundary.
+        """
+        distance_list = []
+        # calculate distance from element to left and lower boundary
+        distance_list.extend(position - self.canvas.position)
+        # calculate distance from element to upper and right boundary
+        distance_list.extend(self.canvas.position + self.canvas.size - position)
+
+        return min(distance_list)
+
+    def draw_shape(self, shape_type, current_position, in_process=False):
+        """Draws the required shape at the given position.
+
+        Parameters
+        ----------
+        shape_type: string
+            Type of shape - line, quad, circle.
+        current_position: (float,float)
+            Lower left corner position for the shape.
+        in_process: bool, optional
+            Checks whether in process or not.
```

Comment:
Not sure of this variable goal... Can you explain?

Reply:
When the `left_mouse_button` is pressed we initiate the drawing of the shape (ie. add the starting point of the shape) then as we want to resize the shape according to the mouse movement, I need to check whether the shape creation is in the process to apply the resizing according to the current mouse position.